projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts

```diff
@@ -7,6 +14,80 @@ import { NavigationNode } from './navigation-node.model';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class NavigationUIComponent {
-  @Input() dropdownMode = 'list';
+  /**
+   * The navigation node to render.
+   */
   @Input() node: NavigationNode;
+
+  /**
+   * the icon type that will be used for navigation nodes
+   * with children.
+   */
+  iconType = ICON_TYPE;
+
+  /**
+   * Indicates whether the navigation should support flyout.
+   * If flyout is set to true, the
+   * nested child navitation nodes will only appear on hover or focus.
+   */
+  @Input() @HostBinding('class.flyout') flyout = true;
+
+  @Input() @HostBinding('class.is-open') isOpen = false;
+
+  private openNodes: HTMLElement[] = [];
+
+  constructor(private router: Router) {
+    this.router.events.subscribe(() => this.clear());
```

Comment:
Maybe we should limit clearing to sth like this:
`events.pipe(filter(event => event instanceof NavigationEnd))`
(or NavigationStart, depending on needs)